tests/src/packages/websocket/WebSocketTests.scala

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2015-2016 IBM Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package packages.websocket
+
+import java.net.URI
+
+import org.junit.runner.RunWith
+import org.scalatest.BeforeAndAfterAll
+import org.scalatest.Finders
+import org.scalatest.junit.JUnitRunner
+
+import common.TestHelpers
+import common.Wsk
+import common.WskProps
+import common.WskTestHelpers
+import spray.json.DefaultJsonProtocol.StringJsonFormat
+import spray.json.pimpAny
+
+@RunWith(classOf[JUnitRunner])
+class WebSocketTests
+        extends TestHelpers
+        with WskTestHelpers
+        with BeforeAndAfterAll {
+
+    implicit val wskprops = WskProps()
+    val wsk = new Wsk()
+    var serverURI: URI = WebSocketUtils.startWebSocketServer()
+
+    val websocketSendAction = "/whisk.system/websocket/send"
+
+    behavior of "Websocket action"
+
+    override def afterAll() {
+        WebSocketUtils.stopWebSocketServer()
+    }
+
+    it should "Stand up a websocket server for testing" in {
+        println(s"Server URI: ${serverURI}")
+
+        var serverStarted = false
+        var retryAttempt = 0
+
+        // The test server starts/runs in the background. Let's give it some time
+        // to finish initializing before declaring that it isn't running.
+        while (!serverStarted && retryAttempt < 3) {
+            retryAttempt = retryAttempt + 1
+
+            println(s"Contacting test websocket server attempt #${retryAttempt}")
+            val testClient = WebSocketUtils.getSocketClient(serverURI)
+            serverStarted = testClient.connectBlocking()
+
+            if (!serverStarted) {
+                Thread.sleep(1000)
+            } else {
+                testClient.closeBlocking()
+            }
+        }
+
+        serverStarted should be(true)
+    }
+
+    it should "Use the websocket action to send a payload" in withAssetCleaner(wskprops) {
+        (wp, assetHelper) =>
+            val run = wsk.action.invoke(websocketSendAction, Map("uri" -> serverURI.toString.toJson, "payload" -> "This is the message to send".toJson))
```

Comment:
The payload _could_ be unique, but even if it were, it would have no effect on the test. In the second use, perhaps changing it to something like `"This payload cannot reach badURI"` might increase readability... a bit... maybe?